lib/client-api/src/story_store.ts

```diff
@@ -452,4 +296,40 @@ export default class StoryStore extends EventEmitter {
   cleanHooksForKind(kind: string) {
     this.getStoriesForKind(kind).map(story => this.cleanHooks(story.id));
   }
+
+  removeStoryKind(kind: string) {
+    this.cleanHooksForKind(kind);
+    this._data = Object.entries(this._data).reduce((acc, [id, story]) => {
+      if (story.kind !== kind) {
+        Object.assign(acc, { [id]: story });
+      }
+      return acc;
+    }, {});
+    this.pushToManager();
+  }
+
+  // This API is a reimplementation of Storybook's original getStorybook() API.
+  // As such it may not behave *exactly* the same, but aims to. Some notes:
+  //  - It is *NOT* sorted by the user's sort function, but remains sorted in "insertion order"
+  //  - It does not include docs-only stories
+  getStorybook(): GetStorybookKind[] {
+    return Object.values(
+      this.raw().reduce((kinds: { [kind: string]: GetStorybookKind }, story) => {
+        if (!includeStory(story)) return kinds;
+
+        const {
+          kind,
+          name,
+          storyFn,
+          parameters: { fileName },
+        } = story;
+
+        if (!kinds[kind]) Object.assign(kinds, { [kind]: { kind, fileName, stories: [] } });
```

Comment:
is this better than:

```js
kinds[kind] = { kind, fileName, stories: [] }
```

Reply:
Not really :) it avoids a linting error (`no-param-reassign`) but it's sort of a cheat. I just did it because I noticed similar code in the codebase. Happy to switch though.

Reply:
If it's all the same, the param reassign seems more readable to me :)